lib/livebook/application.ex

```diff
@@ -19,7 +19,9 @@ defmodule Livebook.Application do
       # Start the server responsible for associating files with sessions
       Livebook.Session.FileGuard,
       # Start the Endpoint (http/https)
-      LivebookWeb.Endpoint
+      LivebookWeb.Endpoint,
+      # Start the Node Pool for managing node names
+      Livebook.Runtime.NodePool
```

Comment:
We should start this before the endpoint. Otherwise the endpoint may accept a request to use the nodepool before the process is started. Unlikely, but possible. :)